fe/fe-core/src/main/java/com/starrocks/connector/delta/DeltaLakeFileStats.java

```diff
@@ -0,0 +1,302 @@
+        if (!hasValidColumnMetrics) {
+            return;
+        }
+
+        updateStats(this.maxValues, newStat, nullCounts, recordCount, predicate);
+    }
+
+    private static Object sumNullCount(Object left, Object value) {
+        return (Double) left + (Double) value;
+    }
+
+    private Optional<Double> getBoundStatistic(String colName, Map<String, Object> boundValues) {
+        if (boundValues == null) {
+            return Optional.empty();
+        }
+        StructField field = schema.get(colName);
+        if (field == null) {
+            return Optional.empty();
+        }
+        Object value = boundValues.get(colName);
+        if (value == null) {
+            return Optional.empty();
+        }
+        return convertObjectToOptionalDouble(field.getDataType(), value);
+    }
+
+    private void updateStats(Map<String, Object> curStat,
+                            Map<String, Object> newStat,
+                            Map<String, Object> nullCounts,
+                            long recordCount,
+                            Predicate<Integer> predicate) {
+        if (newStat == null || nullCounts == null) {
+            hasValidColumnMetrics = false;
+            return;
+        }
+        for (String col : nonPartitionPrimitiveColumns) {
+            if (corruptedStats.contains(col)) {
+                continue;
+            }
+
+            Object newValue = newStat.get(col);
+            if (newValue == null) {
+                Double nullCount = (Double) nullCounts.get(col);
+                if ((nullCount == null) || (nullCount.longValue() != recordCount)) {
+                    curStat.remove(col);
+                    corruptedStats.add(col);
+                }
+                continue;
+            }
+
+            DataType type = schema.get(col).getDataType();
+            Object oldValue = curStat.putIfAbsent(col, newValue);
+            if (oldValue != null) {
+                Comparator<Object> comparator = DeltaLakeComparators.forType(type);
+                if (predicate.test(comparator.compare(oldValue, newValue))) {
+                    curStat.put(col, newValue);
+                }
+            }
+        }
+    }
+
+    public void updateNullCount(Map<String, Object> nullCounts, List<String> nonPartitionPrimitiveColumns) {
+        if (!hasValidColumnMetrics) {
+            return;
+        }
+        if (nullCounts == null) {
+            hasValidColumnMetrics = false;
+            return;
+        }
+
+        for (String col : nonPartitionPrimitiveColumns) {
+            DataType type = schema.get(col).getDataType();
+            if (BasePrimitiveType.isPrimitiveType(type.toString())) {
+                this.nullCounts.merge(col, nullCounts.get(col), DeltaLakeFileStats::sumNullCount);
+            }
+        }
+    }
+
+    private static double parseTimestampWithAtZone(String str) {
+        OffsetDateTime time = OffsetDateTime.parse(str, TIMESTAMP_ZONE_FORMAT);
```

Comment:
could this get the timestamp with zone?

Reply:
Yes, the format is                 

"col_timestamp": "2024-04-27T12:00:00.000+08:00",